modules/adfs/lib/IdP/ADFS.php

```diff
@@ -98,14 +107,14 @@ private static function signResponse($response, $key, $cert)
         $objXMLSecDSig = new XMLSecurityDSig();
         $objXMLSecDSig->idKeys = array('AssertionID');	
         $objXMLSecDSig->setCanonicalMethod(XMLSecurityDSig::EXC_C14N);	
-        $responsedom = \SAML2\DOMDocumentFactory::fromString(str_replace ("\r", "", $response));
+        $responsedom = \SAML2\DOMDocumentFactory::fromString(str_replace("\r", "", $response));
         $firstassertionroot = $responsedom->getElementsByTagName('Assertion')->item(0);
         $objXMLSecDSig->addReferenceList(
             array($firstassertionroot), XMLSecurityDSig::SHA1,
             array('http://www.w3.org/2000/09/xmldsig#enveloped-signature', XMLSecurityDSig::EXC_C14N),
             array('id_name' => 'AssertionID')
         );
-        $objKey = new XMLSecurityKey(XMLSecurityKey::RSA_SHA1, array('type'=>'private'));
+        $objKey = new XMLSecurityKey(XMLSecurityKey::RSA_SHA1, array('type' => 'private'));
```

Comment:
Is SHA1 hardcoded on purpose? Should it be?

Reply:
Well, it should probably be configurable instead of hardcoded, in a similar way we can do this for SAML-entities.. I'll see if I can add this, although it is out of scope for this PR.

Reply:
IIRC, that's the way you initialize `XMLSecurityKey`, and then you _cast_ the key to use the actual intended algorithms...

Reply:
Love it